src/box/memtx_space.c

```diff
@@ -156,6 +156,9 @@ memtx_space_replace_build_next(struct space *space, struct tuple *old_tuple,
 			       enum dup_replace_mode mode,
 			       struct tuple **result)
 {
+	if (memtx_space_on_replace(space) != 0)
```

Comment:
Wouldn't it be better to call this logic in `memtx_engine_recover_snapshot_row` (or somewhere near) instead of patching all replace specializations? Or that's not possible for some reasons?

1. It would be nice to keep all recovery-related things in `memtx_engine.cc` (at least, as much as possible) - I believe that it would be easier to read this code in the future then.
2. Building an index right from replace in another index doesn't break anything but seems confusing (at least, to me)
3. Also, this `memtx_space_on_replace` function seems a bit confusing to me - I start to think about on_replace triggers when reading this name. If we would call this logic only in `memtx_engine_recover_snapshot_row`, we could come up with a better name and this small problem would go away.

Reply:
The problem with only patching the recovery procedure is that we'll have to do it both in the memtx engine (`memtx_engine_recover_snapshot_row`) and applier (`apply_snapshot_row`). That would require using the memtx API in replication, that would look ugly without introducing some virtual methods (that would be empty for other engines).

Renamed the method to `memtx_space_on_replace_build_next`, since it's only called from the `memtx_space_replace_build_next` now.

Reply:
It's not about the sort data, but about changing the recovery precedure.

We've made the PKs be built right after the space's last tuple is loaded from the snapshot (and SKs to be built after that if we have MVCC or sort data enabled).

This required either introducing this call in the replace function or changing both memtx engine and the applier. We could, for sure, continue building PKs the old way when doing replication (not after the space recovery, but after all spaces are loaded from the .snap), but it will make the system use several approaches to building indexes at the same time, which is unnecessary complexity.